ChangeLog.d/reduce-cpu-modifiers-to-file-scope.txt

```diff
@@ -0,0 +1,4 @@
+Bugfix
+   * Fix #5758. Compilers might generate unexpected instructions when CPU
+     modifiers were specified as global flags( command line or global headers).
+     It is fixed with reducing the scope of CPU modifier flags.
```

Comment:
I don't think this is right, at least not since the last release.

I didn't think this could happen in the last release.

I seem to remember the problem happened that once we merged the AESCE code, the compiler would then emit SHA3 instructions, but I can't find the comment on a PR somewhere that explained this clearly. Can you point me at the comment please?

And there was a discussion about whether the user's application needed to be built with the same flags as the library (which is what having everything in `check_config.h` does.

And 5758 is about builds in our CI - the error quoted is `-march=armv8-a+crypto or -march=armv8.2-a+sha3 needed` - that's a problem with `all.sh`